api/src/pcapi/core/users/email/update.py

```diff
@@ -132,6 +132,29 @@ def confirm_email_update_request(token: str) -> None:
         )
 
 
+def cancel_email_update_request(token: str) -> None:
+    """Cancel the email update request for the given user"""
+
+    payload = ChangeEmailTokenContent.from_token(token)
+    current_email = payload.current_email
+    new_email = payload.new_email
+    user = users_repository.find_user_by_email(current_email)
+    if not user:
+        raise exceptions.InvalidEmailError()
+    check_and_expire_token(user, token, TokenType.CONFIRMATION)
+    try:
+        with transaction():
+            models.UserEmailHistory.build_cancellation(user, new_email)
+            api.suspend_account(
+                user, constants.SuspensionReason.FRAUD_HACK, user, "Suspension suite à un changement d'email annulé"
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities.  -->
**💭 thought:** ‏Pas sur qu'on soit sur du hack
J'aurais plutôt mis `FRAUD_SUSPICION` ou même créé un noveau code du style `USER_BLOCKED_EMAIL_CHANGE`

Reply:
j'était partit sur un nouveau truc mais j'ai vu sur figma qu'il faut mettre un motif hacking
mais oui je pense que fraud suspicion est bien. ça dépend du niveau de détails que dont les admins ont besoins, ily a le comentaire pour expliquer la raison.
ducoup je mêt fraud suspicion